### PR TITLE
Adding user authentication needed by latest z-way API. Adding some mo…

### PR DESCRIPTION
**Pull request by thexperiments**
…re debug output.

### CODE REVIEW

File: z-way.coffee [hunk 1]
```diff
@@ -60,12 +70,62 @@ module.exports = (env) ->
...
       env.logger.debug("sending command " + address)
-      return rp(address)
+
+      @logIn()
+      .then ()=>
+        commandAnswer = rp address, @options
+        .then (json) =>
+          try
+            #needed because json could contain circular reference
+            env.logger.debug "received command answer:#{JSON.srtingify(json)}"
```
Comment (MMore) [inline z-way.coffee:80]: Looks like a typo. Have you really tested this version?

**Comment by thexperiments** [comment]
Can't see a typo. Which line? Running since two weeks now and I'm pretty sure I didn't change it after my pull request.

**Comment by MMore** [comment]
I mean the `JSON.srtingify()`. Should be a `JSON.stringify()` I guess.

**Comment by thexperiments** [comment]
Oh yes, maybe  you can correct it after pulling? It is within a try and thus doesn't affect operation that's why I didn't notice

**Comment by MMore** [comment]
Thanks for your work, @thexperiments!

**Pull request merged**